### PR TITLE
[full-ci] Initial implementation of archiver v2

### PR DESCRIPTION
**Pull request by kulmann**
Initial implementation of archiver v2 (REVA) for authenticated contexts. Public links are not supported, yet.

## Related Issue
- Fixes https://github.com/owncloud/web/issues/3913
- Fixes https://github.com/owncloud/web/issues/5809

## Motivation and Context
Being able to download folders

## How Has This Been Tested?

- not yet

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Reva capabilities need to be brought to oCIS https://github.com/owncloud/ocis/pull/2529
- [x] archiver config needs to be merged in oCIS https://github.com/owncloud/ocis/pull/2509 
- [x] updated archiver implementation in reva needs to be merged https://github.com/cs3org/reva/pull/2100
- [x] separate `webUIFilesActionMenu/fileFolderActionMenu.feature:14` into an oCIS and an oC10 test (archiver only exists with ocis)
- [ ] ~~move archiverService initialization to the `initialize` hook once https://github.com/owncloud/web/pull/5874 is merged~~

### CODE REVIEW

File: packages/web-app-files/src/components/AppBar/SelectedResources/BatchActions.vue [hunk 1]
```diff
@@ -24,6 +24,17 @@
       </oc-button>
     </template>
     <oc-grid v-if="displayBulkActions" gutter="small">
+      <div v-if="canDownloadAsArchive">
```
Comment (pascalwengerter) [inline packages/web-app-files/src/components/AppBar/SelectedResources/BatchActions.vue:27]: Currently, if you only have one file selected (e.g. an image), it downloads this image inside a `.tar` file. Can we add a distinction here on whether a single file or a folder / multiple files are selected and handle the download interaction accordingly? :)

Comment (kulmann) [reply packages/web-app-files/src/components/AppBar/SelectedResources/BatchActions.vue:27]: yes, good idea 👍

Comment (kulmann) [reply packages/web-app-files/src/components/AppBar/SelectedResources/BatchActions.vue]: Please check if the implementation reflects your idea properly.

**Comment by ownclouders** [comment]
Results for <strong>oC10SharingInternalUsers</strong> https://drone.owncloud.com/owncloud/web/19381/25/1
The following scenarios passed on retry:
- webUISharingInternalUsersShareWithPage/shareWithUsers.feature:91

**Review by pascalwengerter — APPROVED** [review]
LGTM 🗃️ if CI agrees

**Comment by ownclouders** [comment]
Results for <strong>oC10Files1</strong> https://drone.owncloud.com/owncloud/web/19427/12/1
The following scenarios passed on retry:
- webUIFilesActionMenu/versions.feature:36

**Pull request merged**